main.go

```diff
@@ -106,19 +111,25 @@ var (
 )
 
 // scanDevices uses smartctl to gather the list of available devices.
-func scanDevices(logger log.Logger) []string {
+func scanDevices(logger log.Logger) []Device {
 	filter := newDeviceFilter(*smartctlDeviceExclude, *smartctlDeviceInclude)
 
 	json := readSMARTctlDevices(logger)
 	scanDevices := json.Get("devices").Array()
-	var scanDeviceResult []string
+	var scanDeviceResult []Device
 	for _, d := range scanDevices {
-		deviceName := d.Get("name").String()
+		deviceName := d.Get("info_name").String()
 		if filter.ignored(deviceName) {
 			level.Info(logger).Log("msg", "Ignoring device", "name", deviceName)
 		} else {
 			level.Info(logger).Log("msg", "Found device", "name", deviceName)
-			scanDeviceResult = append(scanDeviceResult, deviceName)
+			device := Device{
+				Name:      d.Get("name").String(),
+				Info_Name: d.Get("info_name").String(),
+				Protocol:  d.Get("protocol").String(),
```

Comment:
device.Protocol is written but never read. Missing code?

Reply:
It's only represents `smartctl --scan --json` to object
I'm not Go-developer)

Reply:
One of the frustrations with `smartctl --scan` is that it can present duplicates  
```
[root@a666 ~]# /opt/index/sbin/smartctl --scan
/dev/sda -d scsi # /dev/sda, SCSI device
/dev/sdb -d scsi # /dev/sdb, SCSI device
/dev/bus/0 -d megaraid,0 # /dev/bus/0 [megaraid_disk_00], SCSI device
/dev/bus/0 -d megaraid,1 # /dev/bus/0 [megaraid_disk_01], SCSI device
/dev/bus/0 -d megaraid,3 # /dev/bus/0 [megaraid_disk_03], SCSI device
[root@a666 ~]#
```
Here `sda` is an HBA VD and `sdb` is a plain drive passed through.

Reply:
but you don't know what of megaraid device is sda
I have filtered scraped devices by exclude /dev/sd*

Reply:
> This is the --scan path, but what about drives where `smartctl --scan` does not list them?
> 
> ```
> $ smartctl   -d 3ware,12 /dev/twl0 -x |head
> smartctl 7.4 2023-08-01 r5530 [x86_64-linux-5.18.5-gentoo-dist] (local build)
> Copyright (C) 2002-23, Bruce Allen, Christian Franke, www.smartmontools.org
> 
> === START OF INFORMATION SECTION ===
> Model Family:     Seagate Barracuda XT
> Device Model:     ST33000651AS
> Serial Number:    (redacted)
> LU WWN Device Id: (redacted)
> Firmware Version: CC44
> User Capacity:    3,000,592,982,016 bytes [3.00 TB]
> ```

I don't have any devices like this.
And I am not an author of exporter
Who are want to maintain - welcome

Reply:
Fair point.  I suggest that this be forked into its own issue and potential PR.  LSI HBAs dominate the server industry so there would be significant benefit from merging this as-is and revisiting later to support less-common devices.

Reply:
How long time to wait merge?

Reply:
> This is the --scan path, but what about drives where `smartctl --scan` does not list them?

That would be an upstream request. I don't think the `smartctl_exporter` should get in the habit of somehow detecting drives to make up for lack of functionality in `smartctl`.

Reply:
... especially if there's a workaround to feed unusual devices to the exporter explicitly.